aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -217,6 +211,48 @@ TORCH_META_FUNC2(mean, dim)
   resize_reduction(*this, self, dim, keepdim, out_dtype);
 }
 
+ScalarType get_result_or_self_value_dtype(
+    const Tensor& self,
+    const Tensor& result,
+    const c10::optional<ScalarType>& dtype) {
+  if (result.defined()) {
+    return result.scalar_type();
+  } else {
+    return dtype.value_or(toValueType(self.scalar_type()));
+  }
+}
+
+void check_norm(const Tensor& self, optional<ScalarType> opt_dtype) {
+  auto in_dtype = opt_dtype.value_or(self.scalar_type());
+
+  TORCH_CHECK(
+      self.layout() == Layout::Strided,
+      "norm only supports strided layout, but got: ", self.layout());
```

Comment:
Technically, now that the functions calling these are no longer composite, this test isn't needed anymore, since sparse will route to SparseCPU and there won't be any kernel.